### ISSUE TITLE
Possible bug in the exemple "UNET 3+ with deep supervision, classification-guided module, and hybrid loss" // n_labels vs input_tensor

### ISSUE THREAD

**Issue by hubsilon** [yingkaisha/keras-unet-collection#61]
dear all,

I'm trying to run the exemple of the Unet3+ (in google colab) on a dataset containing gray images with 3 labels. The given exemple is using the oxford pet dataset, so rgb images (channel=3) with 3 labels.

To do this, I changed the beginning of the code as follow:

n_labels = 3
to
n_labels = 2

and

input_tensor = keras.layers.Input((128, 128, 3))
to
input_tensor = keras.layers.Input((128, 128, 1))

which gives a OUT_stack:
[<KerasTensor: shape=(None, 128, 128, 2) dtype=float32 (created by layer 'unet3plus_output_sup0_activation')>,
 <KerasTensor: shape=(None, 128, 128, 2) dtype=float32 (created by layer 'unet3plus_output_sup1_activation')>,
 <KerasTensor: shape=(None, 128, 128, 2) dtype=float32 (created by layer 'unet3plus_output_sup2_activation')>,
 <KerasTensor: shape=(None, 128, 128, 2) dtype=float32 (created by layer 'unet3plus_output_sup3_activation')>,
 <KerasTensor: shape=(None, 128, 128, 2) dtype=float32 (created by layer 'unet3plus_output_final_activation')>]

the OUT_stack in the oxford pet exemple is:
[<KerasTensor: shape=(None, 128, 128, 3) dtype=float32 (created by layer 'unet3plus_output_sup0_CGM')>,
 <KerasTensor: shape=(None, 128, 128, 3) dtype=float32 (created by layer 'unet3plus_output_sup1_CGM')>,
 <KerasTensor: shape=(None, 128, 128, 3) dtype=float32 (created by layer 'unet3plus_output_sup2_CGM')>,
 <KerasTensor: shape=(None, 128, 128, 3) dtype=float32 (created by layer 'unet3plus_output_sup3_CGM')>,
 <KerasTensor: shape=(None, 128, 128, 3) dtype=float32 (created by layer 'unet3plus_output_final_CGM')>])

I tried a different number of n_labels, for exemple 7 and the OUT_stack becomes:
[<KerasTensor: shape=(None, 128, 128, 7) dtype=float32 (created by layer 'unet3plus_output_sup0_activation')>,
 <KerasTensor: shape=(None, 128, 128, 7) dtype=float32 (created by layer 'unet3plus_output_sup1_activation')>,
 <KerasTensor: shape=(None, 128, 128, 7) dtype=float32 (created by layer 'unet3plus_output_sup2_activation')>,
 <KerasTensor: shape=(None, 128, 128, 7) dtype=float32 (created by layer 'unet3plus_output_sup3_activation')>,
 <KerasTensor: shape=(None, 128, 128, 7) dtype=float32 (created by layer 'unet3plus_output_final_activation')>]

I'd have thought that "shape=(None, 128, 128, 7)" in the OUT_stack is the input_tensor, thus (None, 128, 128, 1) for gray images or (None, 128, 128, 3) for rgb images. Is that a bug or do I  misunderstand something?

**Comment by murdav** [comment]
Try to have a look at this example (lines 8-10):
https://github.com/yingkaisha/keras-vision-transformer/blob/main/examples/Swin_UNET_oxford_iiit.ipynb

I think you have used `n_labels` equals for input and output layers. If so, you need to have hot-encoded mask.
